### ISSUE TITLE
Skylanders Spyro's Adventure [BLES01272] - Skipping shadows visual bug with water still present

### ISSUE THREAD

**Issue by EspeonScottie** [RPCS3/rpcs3#13248]
### Quick summary

Came back to this after a while to see if it was still happening, and on the latest build I tested (0.0.26-14577) it seems so.

### Details

Happens in various levels with water. Original issue was identified in #12987, and a fix #13097 was supposedly implemented but hasn't seemed to have worked for this game.
![Capture](https://user-images.githubusercontent.com/77664800/213011381-45278f9e-dc50-46e0-9ff0-07504a895294.PNG)

### Attach a log file

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/10439296/RPCS3.log.gz)

### Attach capture files for visual issues

[BLES01272_20230117154442_capture.zip](https://github.com/RPCS3/rpcs3/files/10439298/BLES01272_20230117154442_capture.zip)

### System configuration

Intel Core i7-6700HQ
NVIDIA GeForce GTX 1060 (527.56)
Windows 10

**Comment by killedmegaman** [comment]
I have the same problem, but it appears fron the second level. 
The game its dizzy and the preformance drops

**Comment by Kr0m4t1c** [comment]
Same issue here

**Comment by hexax1** [comment]
same problem here too

**Comment by Duratlas** [comment]
I've also been having this problem, although it does not drop performance for me. Stays consistently at 30fps. I'm on a laptop with Ryzen 7 6800H 3.2GHz, RTX3060 6GB VRAM, and 16GB RAM.

**Comment by fede242rico** [comment]
same problem

**Comment by LeMustelide** [comment]
Has anyone found a solution yet?

**Comment by Schmumschmenz** [comment]
hey guys, I had the same problem with the second level and was looking for an answer but couldnt find s**t... but I managed to fix it for me by downgarding to an older version, I was just looking up generall tutorials on youtube for setting the game up. but they didnt mention any other bug or glitch like this, soo I just tried to use the older version that he used intead!

I got version 0.0.24-14340 and replaced it in my emulators directory and that fixed it for me (for now atleast)
here is the link of the offical download page for older releases, the version taht I used is the first one on top: https://rpcs3.net/compatibility?b&p=20

I hope thatl fix it for anyone else.

**Comment by kd-11** [comment]
If it's a regression maybe someone can help figure out when it broke so we can fix it faster. It would be a great help.

**Comment by Graubek** [comment]
Seems like https://github.com/RPCS3/rpcs3/pull/13201 is responsible for it

Footage from 0.0.26-14542
![rpcs3-v0 0 26-14542-659ee81e_win64](https://github.com/RPCS3/rpcs3/assets/58905125/aff26959-c307-4bab-9c87-9a1d48148d18)

Footage from 0.0.26-14564
![0 0 26-14564](https://github.com/RPCS3/rpcs3/assets/58905125/e0f1bf02-7b4f-401e-961b-786416f8b678)

**Comment by Graubek** [comment]
> If it's a regression maybe someone can help figure out when it broke so we can fix it faster. It would be a great help.

> Seems like #13201 is responsible for it
> 
> Footage from 0.0.26-14542 ![rpcs3-v0 0 26-14542-659ee81e_win64](https://user-images.githubusercontent.com/58905125/277989856-aff26959-c307-4bab-9c87-9a1d48148d18.png)
> 
> Footage from 0.0.26-14564 ![0 0 26-14564](https://user-images.githubusercontent.com/58905125/277990483-e0f1bf02-7b4f-401e-961b-786416f8b678.png)

**Comment by epicEaston197** [comment]
I'm also experiencing this issue 
Started happening to me after the second level like others have stated But oddly enough it is completely not present in the hub world

![Image](https://github.com/user-attachments/assets/04c43077-81ac-42c8-9264-e3527b505b49)
![Image](https://github.com/user-attachments/assets/d5c8791d-045f-420e-9634-e78fa6a0b57d)

**Comment by NanoViridescence** [comment]
I would like to say that setting the shader quality to low worked specifically for me on the latest version, thanks reddit. Figured I'd share.

**Comment by digant73** [comment]
reported as regressed by #13201

**Comment by digant73** [comment]
following and comparing the PRs reported as fixing and regressing this game, I see:

- #13097 (fixing the game) simply made some changes on function (in `rsx_methods.cpp`):

`void set_surface_format(thread* rsx, u32 reg, u32 arg)`

- #13201 (regressing the game) made some changes on the same function. In particular, the implementation of method `antialias()` (used on the `if` statement) is the only one that has been changed.

On #13201 (regressed code), `antialias()` is calling a method `rsx::to_surface_antialiasing(u8 in)` that checks on a range of values:
```
static inline auto to_surface_antialiasing(u32 in)
{
	return gcm_enum_cast<surface_antialiasing>(in,
	{
		{ CELL_GCM_SURFACE_CENTER_1, CELL_GCM_SURFACE_CENTER_1 },
		{ CELL_GCM_SURFACE_DIAGONAL_CENTERED_2, CELL_GCM_SURFACE_SQUARE_ROTATED_4 }
	});
}
```

On #13097 (not regressed code), `antialias()` is calling a method `rsx::to_surface_antialiasing(u8 in)` that applies a switch statement returning different output values:
```
surface_antialiasing rsx::to_surface_antialiasing(u8 in)
{
	switch (in)
	{
	case CELL_GCM_SURFACE_CENTER_1: return surface_antialiasing::center_1_sample;
	case CELL_GCM_SURFACE_DIAGONAL_CENTERED_2: return surface_antialiasing::diagonal_centered_2_samples;
	case CELL_GCM_SURFACE_SQUARE_CENTERED_4: return surface_antialiasing::square_centered_4_samples;
	case CELL_GCM_SURFACE_SQUARE_ROTATED_4: return surface_antialiasing::square_rotated_4_samples;
	}
	return throw_exception_if_emulating("Unknown surface antialiasing format 0x%x", in);
}
```

Maybe the issue could be on that different method's logic and could be easily fixed by @kd-11

**Comment by adina-the-nerd** [comment]
Issue still present. Furthermore random freezing

**Comment by digant73** [comment]
@adina-the-nerd did you test on latest build 0.0.39-18602?

**Comment by adina-the-nerd** [comment]
@digant73  Yep, the crashing & freezing at 1440p was fixed (I think) but not the striped shadows

**Comment by adina-the-nerd** [comment]
<img width="2559" height="1439" alt="Image" src="https://github.com/user-attachments/assets/62648ec8-2299-432c-ac29-ba9161a0c0c8" />

<img width="1280" height="1024" alt="Image" src="https://github.com/user-attachments/assets/279d4c23-1b46-4591-a3af-cf5951649ff0" />

**Comment by digant73** [comment]
@adina-the-nerd attach the log file please. Try also `Shader Quality` set to LOW

**Comment by kd-11** [comment]
This one needs a fresh RSX capture of a scene where the issue is close to the camera and obvious. I'm going through all the regressions over the next few weeks.

**Comment by ieris19** [comment]
> This one needs a fresh RSX capture of a scene where the issue is close to the camera and obvious. 

I can potentially try to go through and get whatever info you need, but I would need to know how to do this. Is there any place I can read about how to do an RSX capture?

**Comment by kd-11** [comment]
> > This one needs a fresh RSX capture of a scene where the issue is close to the camera and obvious.
> 
> I can potentially try to go through and get whatever info you need, but I would need to know how to do this. Is there any place I can read about how to do an RSX capture?

Just play the game and when the issue appears on screen just click on "Utilities -> Create RSX Capture" (there's also a keyboard shortcut but I cannot remember)

**Comment by epicEaston197** [comment]
> I'm also experiencing this issue Started happening to me on the second level like others have stated But oddly enough it is completely not present in the hub world
>
>I never made it past this point because of how visually distracting it is

I decided to launch the game to get logs and such for but I found out that the issue is completely fixed with the latest version I don't know if it was fixed with previous versions but it's working now

And I don't recall changing any settings

<img width="1919" height="1079" alt="Image" src="https://github.com/user-attachments/assets/facc897f-875b-4071-b1cb-87e9ffb8325d" />

**Comment by digant73** [comment]
@epicEaston197 if fixed recently, the issue was probably fixed by 0.38-18577 (#17862). If possible, please try that build and the previous one 0.0.38-18556 (#17916) and confirm my suspect (I just made some testing (also changing some code here and there) for other issues on other games and that build was changing something).

@adina-the-nerd, @ieris19 please try the current build first and let us know if the issue is fixed also for you. If not, provide the RSX file as explained by kd-11

**Comment by kd-11** [comment]
Closing as fixed as per last evidence provided.
